experimental/teams-tabs/javascript_nodejs/bots/teamsAdaptiveCardTabsBot.js

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActionTypes, TeamsActivityHandler, MessageFactory } = require('botbuilder');
+const { CardResources } = require('../models/cardResources');
+const { UIConstants } = require('../models/UIConstants');
+const { TaskModuleResponseFactory } = require('../models/taskModuleResponseFactory');
+const { SimpleGraphClient } = require('../simpleGraphClient');
+const path = require('path');
+
+class TeamsAdaptiveCardTabsBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+
+        this.connectionName = process.env.connectionName;
+
+        // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
+        this.onMessage(async (context, next) => {
+            if (context.activity?.text?.trim().toLowerCase() === 'logout') {
+                if (typeof context.adapter.signOutUser !== 'function') {
+                    throw new Error('logout: not supported by the current adapter.');
+                }
+
+                await context.adapter.signOutUser(context, this.connectionName);
+                await context.sendActivity('You have been signed out.');
+            } else {
+                await context.sendActivity('Hello, I am a Teams Adaptive Card Tabs bot. Please use the tabs to interact. (Send "logout" to sign out)');
+            }
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+    };
+
+    async handleTeamsTabFetch(context, tabRequest) {
+        if (tabRequest?.tabEntityContext?.tebEntityId === 'workday') {
+            let magicCode;
+            if (tabRequest.state) {
+                magicCode = tabRequest.state.toString();
+            }
+
+            return this.getPrimaryTabResponse(context, magicCode);
```

Comment:
I believe this should be awaited

Reply:
[JavaScript is just weird](https://eslint.org/docs/rules/no-return-await)